src/main/java/org/broadinstitute/dsm/util/JavaHeapDumper.java

```diff
@@ -0,0 +1,84 @@
+package org.broadinstitute.dsm.util;
+
+import com.sun.management.HotSpotDiagnosticMXBean;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.util.GoogleBucket;
+import org.broadinstitute.dsm.statics.ApplicationConfigConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.management.MBeanServer;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class JavaHeapDumper {
+
+    private static final Logger logger = LoggerFactory.getLogger(JavaHeapDumper.class);
+
+    public static final String DEFAULT_LOCAL_PATH = "/tmp";
+    public static final String DEFAULT_BUCKET_PATH = "heap_dumps";
+
+    public void dumpHeapToLocalFile(String filePath) throws IOException {
+        MBeanServer server = ManagementFactory.getPlatformMBeanServer();
+        HotSpotDiagnosticMXBean mxBean = ManagementFactory.newPlatformMXBeanProxy(
+                server, "com.sun.management:type=HotSpotDiagnostic", HotSpotDiagnosticMXBean.class);
+        logger.info("About to create local file at:" + filePath);
+        mxBean.dumpHeap(filePath, true);
+    }
+
+    public void dumpHeapToBucket(String bucketName) throws IOException {
+        dumpHeapToBucket(bucketName, generateDumpFileName());
+    }
+
+    public void dumpHeapToBucket(String bucketName, final String fileName) throws IOException {
+        String gcpName = TransactionWrapper.getSqlFromConfig(ApplicationConfigConstants.GOOGLE_PROJECT_NAME);
+        dumpHeapToLocalFile(DEFAULT_LOCAL_PATH + "/" + fileName);
+        File localDumpFile = Paths.get(DEFAULT_LOCAL_PATH, fileName).toFile();
+        if (localDumpFile.exists()) {
+            logger.info("Created local dump file: " + localDumpFile.getAbsolutePath() + " with size: " + localDumpFile.length());
+        } else {
+            logger.error("Could not find dump file at:" + DEFAULT_LOCAL_PATH + "/" + fileName);
```

Comment:
Probably don't need the logger.error since the next line throws an exception with similar info.